clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java

```diff
@@ -556,6 +638,74 @@ public String toString() {
         public int sizeOf(Object o) {
             return ByteUtils.sizeOfVarlong((Long) o);
         }
+
+        @Override
+        public String documentation() {
+            return "Represents an integer between -2<sup>63</sup> and 2<sup>63</sup>-1 inclusive. " +
+                    "Encoding follows the variable-length zig-zag encoding from " +
+                    " <a href=\"http://code.google.com/apis/protocolbuffers/docs/encoding.html\"> Google Protocol Buffers</a>.";
+        }
     };
 
+    private static String toHtml() {
+        Type dummyType = new Type() {
```

Comment:
Seems we can use any type? Why not reuse STRING or one of the other primitives?

Reply:
If I used String, toString of ArrayOf would return ARRAY(STRING) which would look strange at least in the table. That's why I created a Type with toString returning "T". 
But with the introduction of DocumentedType I think I found a better solution.